### PR TITLE
[13.0][PORT] 661 from 12.0

### PR DESCRIPTION
**Pull request by sebastienbeau**
Port #661 from 12.0 to 13.0.

### CODE REVIEW

File: shopinvader/data/ir_cron.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <record id="shopinvader_sale_order_price_recompute" model="ir.cron" forcecreate="True">
+        <field name="name">Shopinvader - Recompute prices on carts</field>
+        <field name="active" eval="True"/>
+        <field name="user_id" ref="base.user_root"/>
+        <field name="interval_number">1</field>
+        <field name="interval_type">days</field>
+        <field name="numbercall" eval="-1"/>
+        <field name="doall" eval="False"/>
+        <field name="model_id" ref="model_shopinvader_backend"/>
+        <field name="code">model._launch_sale_price_update()</field>
```
Comment (simahawk) [inline shopinvader/data/ir_cron.xml:14]:
```suggestion
        <field name="state">code</field>
        <field name="code">model._launch_sale_price_update()</field>
```

File: shopinvader/models/sale.py [hunk 2]
```diff
@@ -112,6 +112,43 @@ def write_with_onchange(self, vals):
...
 
+    @api.multi
+    def _update_pricelist_and_update_line_prices(self):
+        """
+        On current sale orders (self):
+        - Update the pricelist with the one set on the backend;
+            => If no pricelist, use the default define on the partner.
+        - Then launch the price update (module sale_order_price_recalculation).
+        :return: bool
+        """
```
Comment (simahawk) [inline shopinvader/models/sale.py:123]:
```suggestion
        """Update prices on current sale orders.

        Behavior:

        * Update the pricelist with the one set on the backend;
           => If no pricelist, use the default define on the partner.
        * Then launch the price update (module sale_order_price_recalculation).

        :return: bool
        """
```

File: shopinvader/models/shopinvader_backend.py [hunk 3]
```diff
@@ -426,3 +428,58 @@ def _normalize_name(name):
     def _onchange_name(self):
         if self.name and not self.tech_name:
             self.tech_name = self._normalize_name(self.name)
+        auth_api_key = getattr(request, "auth_api_key", None)
+        return self.browse(self._get_id_from_auth_api_key(auth_api_key))
+
+    @api.multi
```
Comment (simahawk) [inline shopinvader/models/shopinvader_backend.py:434]: multi is gone

File: shopinvader/tests/test_sale.py [hunk 4]
```diff
@@ -152,3 +153,122 @@ def test_invoice_02(self):
                 }
             ],
         )
+
+    def _create_sale_order(self):
+        """
+        Create a new sale.order with 1 line.
+        :return: bool
+        """
+        self.sale = sale = self.env["sale.order"].create(
```
Comment (simahawk) [inline shopinvader/tests/test_sale.py:162]: you should return `sale` instead of setting it on the instance

File: shopinvader/tests/test_sale.py [hunk 5]
```diff
@@ -152,3 +153,122 @@ def test_invoice_02(self):
...
+                        "base_pricelist_id": self.first_pricelist.id,
+                        "date_start": fields.Date.today(),
+                        # TODO: remove timedelta after Odoo date bug
+                        # https://github.com/odoo/odoo/pull/51967
+                        "date_end": fields.Date.today() + timedelta(days=1),
+                    },
+                )
+            ],
+        }
+        self.second_pricelist = self.env["product.pricelist"].create(
```
Comment (simahawk) [inline shopinvader/tests/test_sale.py:271]: return the pricelist instead

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by simahawk** [comment]
duplicate of https://github.com/shopinvader/odoo-shopinvader/pull/679 ?
CC @rousseldenis

**Comment by rousseldenis** [comment]
> duplicate of #679 ?
> CC @rousseldenis

Indeed and depending on a long debate https://github.com/OCA/sale-workflow/pull/1155

About how to trigger onchanges. Form() that I don't like or onchange_helper

**Comment by sebastienbeau** [comment]
I am closing this PR as it a duplicated sorry

**Pull request closed without merging**